themes/default/content/challenge/holiday-shopping.md

```diff
@@ -0,0 +1,268 @@
+                    container_port=8000,
+                    host_port=8000
+                )],
+            ),
+        }
+    ),
+    desired_count=1
+)
+
+# AWS export
+pulumi.export("app url", Output.concat("http://", lb.load_balancer.dns_name))
+```
+
+Run `pulumi up` and select yes to perform the update (this can take up to 10 minutes). Pulumi will output the `app url`. Go to the URL, and you will be presented with Pulumipus' Grocery List web application. You can try to add some items, but nothing will happen because there is no database connected yet. You will do that in the next step.
+
+#### Step 3. Create a MongoDB Atlas database
+
+The web app stores the grocery list items inside a MongoDB database. MongoDB Atlas is a fully managed cloud database that you will be using for this application. Add the below before the code from the previous steps (above the ECR repo declaration) in `__main__.py`. This code creates a MongoDB project, sets up the network access controls, creates a Free Tier cluster in `US_WEST_2` in AWS, and a database user that is scoped to the created cluster.
+
+```python
+# MongoDB Atlas configs
+db_username = config.require("dbUser")
+db_password = config.require_secret("dbPassword")
+atlas_org_id = config.require("orgID")
+
+# Create MongoDB Project
+mongo_project = mongodb.Project("mongo_project", org_id=atlas_org_id)
+
+# Open access to all IPs
+mongo_acl = mongodb.ProjectIpAccessList("mongo_acl",
+    cidr_block="0.0.0.0/0",
+    comment="Open access for backend",
+    project_id=mongo_project.id,)
+
+# Create Free Tier cluster
+mongo_cluster = mongodb.Cluster("mongo-cluster",
+    backing_provider_name="AWS",
+    project_id=mongo_project.id,
+    provider_instance_size_name="M0",
+    provider_name="TENANT",
+    provider_region_name="US_WEST_2")
+
+# Create Database user and give access to the cluster and database
+mongo_user = mongodb.DatabaseUser("db_user",
+    auth_database_name="admin",
+    labels=[mongodb.DatabaseUserLabelArgs(
+        key="project",
+        value="pulumi",
+    )],
+    password=db_password,
+    project_id=mongo_project.id,
+    roles=[
+        mongodb.DatabaseUserRoleArgs(
+            database_name="grocery-list",
+            role_name="readWrite",
+        ),
+        mongodb.DatabaseUserRoleArgs(
+            database_name="admin",
+            role_name="readAnyDatabase",
+        ),
+    ],
+    scopes=[
+        mongodb.DatabaseUserScopeArgs(
+            # Extracts the cluster name to add to database scopes
+            name=Output.all(mongo_cluster.srv_address).apply(lambda v: re.split("\.|\/\/", v[0])[1]),
+            type="CLUSTER",
+        ),
+    ],
+    username=db_username
+)
+
+# MongoDB Atlas export
+pulumi.export("mongo connection string",
+    Output.format("mongodb+srv://{0}:{1}@{2}", db_username, db_password,
+        Output.all(mongo_cluster.srv_address).apply(lambda v: v[0].split("//"))[1])
+)
+```
```

Comment:
It doesn't look like this export is referenced -- should we be exporting it? The value is currently masked with `[secret]` anyway, so unless we explain that you need to `pulumi stack output --show-secrets` to see it, it won't be clear what it's there for. 

If we do want to export it though, it's generally better to use a contiguous string, just do you don't have to surround the values you want to read with quotes:

```suggestion
# MongoDB Atlas export
pulumi.export("mongo_connection_string",
    Output.format("mongodb+srv://{0}:{1}@{2}", db_username, db_password,
        Output.all(mongo_cluster.srv_address).apply(lambda v: v[0].split("//"))[1])
)
```
```

Reply:
This has just been removed since it isn't being used